### PR TITLE
Take care of 'expectation' issues

### PR DESCRIPTION
**Pull request by Sperling-0**
- This work is a part of an upgrade to phpstan level 9

Ticket: https://phabricator.wikimedia.org/T359803

### CODE REVIEW

File: src/Authentication/OldStyleTokens/AccessTokenUrlAuthenticator.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public function addAuthenticationTokensToApplicationUrl( string $url ): string {
 	 * @param class-string $urlGeneratorClass
 	 * @param string[] $requestedParameters
 	 *
-	 * @return array<string, int|string>
+	 * @return array<string, bool|int|string>
```
Comment (gbirke) [inline src/Authentication/OldStyleTokens/AccessTokenUrlAuthenticator.php:29]: The boolean comes from the `json_encode` and is confusing/misleading. Please add `JSON_THROW_ON_ERROR` instead to ensure `custom` is a always string

File: src/Infrastructure/Translation/TranslatorInterface.php [hunk 2]
```diff
@@ -8,7 +8,7 @@ interface TranslatorInterface {
 
 	/**
 	 * @param string $messageKey
-	 * @param array<string, scalar> $parameters
+	 * @param array<string, string|float|null> $parameters
```
Comment (gbirke) [inline src/Infrastructure/Translation/TranslatorInterface.php:11]:
```suggestion
	 * @param array<string, scalar|null> $parameters
```

If it was scalar before, I'd rather not risk an error when passing an integer or boolean.

File: src/Validation/FieldTextPolicyValidator.php [hunk 3]
```diff
@@ -15,7 +15,7 @@ class FieldTextPolicyValidator {
 	public function __construct( private readonly TextPolicyValidator $textPolicyValidator ) {
 	}
 
-	public function validate( $value ): ValidationResult {	// @codingStandardsIgnoreLine
+	public function validate( string $value ): ValidationResult {	// @codingStandardsIgnoreLine
```
Comment (gbirke) [inline src/Validation/FieldTextPolicyValidator.php:18]: This class is entirely unused outside of its test. Please delete it and its test.

**Review by gbirke — APPROVED** [review]

File: src/Infrastructure/Translation/TranslatorInterface.php (hunk 2 above)
Comment (Sperling-0) [reply src/Infrastructure/Translation/TranslatorInterface.php:11]: @gbirke PHPStan was giving me an error on this line, to fix it I made the changes.

If I change it back to `scalar` the error will come back, what to do in that case??

Comment (Sperling-0) [reply src/Infrastructure/Translation/TranslatorInterface.php]: I have made the changes suggested by you.

**Pull request merged**